### DISCUSSION TITLE
New - Can SO be Installed in Azure VM w/o Marketplace Version

### DISCUSSION THREAD

**Discussion by amurray22** [Security-Onion-Solutions/securityonion#5991] [Unsupported Versions] [answered]
I will be  in a college cyber competition that will use Azure as the platform. We will be basically a blue team. Would like to install Security Onion as one of our tools but are only allowed free/open source tools. It looks like installing the Security Onion VM in the Azure Market Place includes fees. Is there a way to create a Linux based VM and then install SO in that VM?

Also saw in the existing Azure installation instructions about the tap issue. Would this also require extra fees to setup taps from the Azure virtual network?

Thanks in advance.

**Comment by jertel** [comment, marked as answer]
Yes, it's possible to run Security Onion in Azure without the official Azure image. The image exists for those that prefer to click and go, have confidence that what they're using has been fully tested for each release, and not have to figure out various cloud compatibility issues. But if you're up for it, there's nothing preventing you from installing it on a new linux VM, and getting it working on your own. 

That said, you will still have the Azure VM and storage resource fees to pay so it won't be completely free. However, Azure does provide a small set of credits to new accounts. To go along with that, the Security Onion image provides a free trial period where there are no charges for the first month of use. So you could run Security Onion for free, provided your competition is about a month or less.

As far as the virtual tap issue, yes that will also involve fees since you will need to utilize a third-party packet broker service on the Marketplace. Though, if you're planning to forward sensor packets into the cloud from a physical node outside of Azure then that could be an option to avoid using a packet broker. It will depend on what systems you're trying to monitor. 

The AWS Security Onion image might be worth looking at, instead of Azure, due to the virtual tap issue.

**Reply by amurray22** [reply]
Thanks for the details. The competition is completely inside an Azure network setup, so we would be looking to send data from this pre-built virtual network to another VM in this same Azure system. 

Just to confirm, if the SO is on this same virtual network as the other systems we want to monitor do we still need the packet broker to move the data from the Azure VMs and network to other Azure VM running SO?

I'll have to clarify with the competition what services we get to use and how 'free and open source' is defined. 

Appreciate the help. I'm excited to get into the details of using the tools.

**Reply by jertel** [reply]
Something needs to be monitoring the network traffic, even if it's in the same virtual network. It sounds like this is a controlled event, so Azure Network Watcher might be an option for you. It can capture for up to 5h. Then perhaps use `so-import-pcap` or `so-tcpreplay` to get the data into Security Onion.

**Reply by amurray22** [reply]
Ok, that makes sense. Appreciate the information.
